app/src/main/java/school/cactus/succulentshop/SignupActivity.kt

```diff
@@ -0,0 +1,58 @@
+package school.cactus.succulentshop
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.textfield.TextInputLayout
+import school.cactus.succulentshop.databinding.ActivitySignupBinding
+
+class SignupActivity :AppCompatActivity() {
+    private lateinit var binding: ActivitySignupBinding
+
+    private val signupEmailValidator = SignupEmailValidator()
+    private val signupIdentifierValidator = SignupIdentifierValidator()
+    private val signupPasswordValidator = SignupPasswordValidator()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignupBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        supportActionBar?.title = getString(R.string.sign_up)
+
+
+
+        binding.apply {
+            signUpButton.setOnClickListener {
+                signupEmailInputLayout.validate()
+                signupUsernameInputLayout.validate()
+                signupPasswordInputLayout.validate()
+            }
+
+            haveAnAccountButton.setOnClickListener {
+                navigateToLoginActivity()
+            }
+        }
+
+    }
+
+    private fun navigateToLoginActivity() {
+        val intent = Intent(this,LoginActivity::class.java)
```

Comment:
Formatting code before committing is a really good practice. You can enable "format code before committing" preference in the Version Control panel.